bin/main.rb

```diff
@@ -1,70 +1,227 @@
   end
 
-  puts "<<<<<slot is #{slots} >>>>"
-  slots == 9 ? start_game = false : true
+  def canvas_rules
+    user_info
+    while @game_on
+      print "#{@player_one.name} its your turn"
+      puts
+
+      @player_one.choice = gets.chomp
+      if validate_player_choice(@player_one.choice)
+        check = true
+        if Integer(@player_one.choice) == @arr[0][0]
+          @arr[0][0] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 1 && @arr[0][0].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[0][1]
+          @arr[0][1] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 2 && @arr[0][1].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[0][2]
+          @arr[0][2] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 3 && @arr[0][2].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[1][0]
+          @arr[1][0] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 4 && @arr[1][0].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[1][1]
+          @arr[1][1] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 5 && @arr[1][1].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[1][2]
+          @arr[1][2] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 6 && @arr[1][2].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[2][0]
+          @arr[2][0] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 7 && @arr[2][0].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[2][1]
+          @arr[2][1] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 8 && @arr[2][1].class == String
+          check = false
+        elsif Integer(@player_one.choice) == @arr[2][2]
+          @arr[2][2] = 'X'
+          check = true
+        elsif Integer(@player_one.choice) == 9 && @arr[2][2].class == String
+          check = false
+        end
+
+        puts 'Slot Occupied, Choose another slot' if check == false
+
+        check == false ? next : true
+
+        if player_win_cases('X')
+          @arr.each do |i|
+            print i
+            puts
+          end
+          player_win_state(@player_one.name)
+          return
+        end
+
+        @arr.each do |i|
+          print i
+          puts
+        end
+        @count += 1
+      elsif !validate_player_choice(@player_one.choice)
+        puts 'Enter a valid number'
+        next
+      end
+
+      if @count == 9
+        @game_on = false
+        break
+      end
+
+      play_second
+      if player_win_cases('O')
+        player_win_state(@player_two.name)
+        return
+      end
+    end
+    puts 'Its a draw'
+  end
```

Comment:
#### Implementation details

- [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing).

both methods **canvas_rules** and **play_second** have several responsibilities, break them down into single responsibility methods

- [ ] Keep your code DRY (Don't repeat yourself)

both methods **canvas_rules** and **play_second**, do similar things, I think you only need one of them and only send a parameter to know it's the next player's turn